cli/tsc/99_main_compiler.js

```diff
@@ -698,6 +698,10 @@ delete Object.prototype.__proto__;
           /** @type {Record<string, any[]>} */
           const diagnosticMap = {};
           for (const specifier of request.specifiers) {
+            if (core.opSync("op_is_cancelled", {})) {
+              break;
+            }
+
             diagnosticMap[specifier] = fromTypeScriptDiagnostic([
               ...languageService.getSemanticDiagnostics(specifier),
               ...languageService.getSuggestionDiagnostics(specifier),
```

Comment:
The language service has a cancellation API, but it doesn't really work like how I would expect a cancellation API to work. It migth be good if we use it in the future. I've asked some questions on the TS discord about it.

Reply:
Actually, I'm just going to implement using their cancellation API this with the throttling. I'll check more what they do in tsserver.